### PR TITLE
cv_version: ensure version param has a minor part

### LINKED ISSUE #1087: content_view_version does not allow creating version numbers > 4 digits, even though katello accepts any valid integer
##### SUMMARY
When publishing a new content view version, and you provide a version number you want it created as, it will not accept more than 4 digits.

I have used versions such as 20201201 in katello for quite a while via hammer and the API.

Something in the collection code is rejecting creating content views with anything more than 4 digit version numbers.

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION
ansible 2.9.14

##### KATELLO/FOREMAN VERSION
3.15

##### STEPS TO REPRODUCE
specify "version: 12345" when creating a new content_view_version

##### EXPECTED RESULTS
Katello/Foreman actually will except anything that is an integer for the major and minor numbers.

##### ACTUAL RESULTS
```
fatal: [localhost -> 127.0.0.1]: FAILED! => {"changed": false, "module_stderr": "Shared connection to 127.0.0.1 closed.\r
", "module_stdout": "/usr/lib/python2.7/site-packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html\r
  InsecureRequestWarning)\r
/usr/lib/python2.7/site-packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html\r
  InsecureRequestWarning)\r
/usr/lib/python2.7/site-packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html\r
  InsecureRequestWarning)\r
/usr/lib/python2.7/site-packages/urllib3/connectionpool.py:769: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html\r
  InsecureRequestWarning)\r
Traceback (most recent call last):\r
  File \"/home/myusername/.ansible/tmp/ansible-tmp-1607103112.51-31237-205164981239553/AnsiballZ_content_view_version.py\", line 102, in <module>\r
    _ansiballz_main()\r
  File \"/home/myusername/.ansible/tmp/ansible-tmp-1607103112.51-31237-205164981239553/AnsiballZ_content_view_version.py\", line 94, in _ansiballz_main\r
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r
  File \"/home/myusername/.ansible/tmp/ansible-tmp-1607103112.51-31237-205164981239553/AnsiballZ_content_view_version.py\", line 40, in invoke_module\r
    runpy.run_module(mod_name='ansible_collections.theforeman.foreman.plugins.modules.content_view_version', init_globals=None, run_name='__main__', alter_sys=True)\r
  File \"/usr/lib64/python2.7/runpy.py\", line 176, in run_module\r
    fname, loader, pkg_name)\r
  File \"/usr/lib64/python2.7/runpy.py\", line 82, in _run_module_code\r
    mod_name, mod_fname, mod_loader, pkg_name)\r
  File \"/usr/lib64/python2.7/runpy.py\", line 72, in _run_code\r
    exec code in run_globals\r
  File \"/tmp/ansible_theforeman.foreman.content_view_version_payload_r83Wpo/ansible_theforeman.foreman.content_view_version_payload.zip/ansible_collections/theforeman/foreman/plugins/modules/content_view_version.py\", line 234, in <module>\r
  File \"/tmp/ansible_theforeman.foreman.content_view_version_payload_r83Wpo/ansible_theforeman.foreman.content_view_version_payload.zip/ansible_collections/theforeman/foreman/plugins/modules/content_view_version.py\", line 213, in main\r
IndexError: list index out of range\r
", "msg": "MODULE FAILURE
See stdout/stderr for the exact error", "rc": 1}
```

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: plugins/modules/content_view_version.py [hunk 1]
```diff
@@ -204,6 +204,13 @@ def main():
 
     module.task_timeout = 60 * 60
 
+    if 'version' in module.foreman_params and '.' not in module.foreman_params['version']:
```
Comment (mdellweg) [inline plugins/modules/content_view_version.py]: What if you get "2.3.4.5"? Should we check for exactly one '.'?

Comment (evgeni) [reply plugins/modules/content_view_version.py]: So something like:

```python
    if 'version' in module.foreman_params and module.foreman_params['version'].count('.') != 1:
        err_msg = "The 'version' needs to be in the format 'X.Y', not '{0}'".format(module.foreman_params['version'])
        if module.foreman_params['version'].count('.') > 1:
            module.fail_json(err_msg)
        else:
            try:
                major_version = int(module.foreman_params['version'])
                module.foreman_params['version'] = "{0}.0".format(major_version)
            except ValueError:
                module.fail_json(err_msg)
```

still doesn't catch "what.lol" as a version, but I'd not go that far right now?

File: plugins/modules/content_view_version.py [hunk 2]
```diff
@@ -204,6 +204,13 @@ def main():
...
+        try:
+            major_version = int(module.foreman_params['version'])
+            module.foreman_params['version'] = "{0}.0".format(major_version)
```
Comment (evgeni) [inline plugins/modules/content_view_version.py:211]: should this branch emit a warning, so the user *knows* their playbook  is not exactly correct?

Comment (mdellweg) [reply plugins/modules/content_view_version.py:211]: if hammer accepts it we should maybe also.

File: plugins/modules/content_view_version.py (hunk 1 above)
Comment (mdellweg) [reply plugins/modules/content_view_version.py]: i thought checking a regex like `r"^(\d*)(?:\.(\d*))?$"`

Comment (evgeni) [reply plugins/modules/content_view_version.py]: I've just used `^\d+\.\d+$` and then relied on `int()` throwing a `ValueError` ;)

**Review by mdellweg — APPROVED** [review]
I needed to read the logic twice, but i think it covers all cases now.

**Pull request merged**